.github/workflows/pr-frontend-unit-tests.yml

```
@@ -31,100 +31,115 @@ jobs:
-        chunk: [1, 2, 3, 4, 5, 6, 7, 8]
+        shard: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
+        total: [16]
     steps:
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
-    - uses: actions/setup-node@v4
-      with:
-        node-version-file: '.nvmrc'
-        cache: 'yarn'
-        cache-dependency-path: 'yarn.lock'
     - name: Setup Enterprise
       uses: ./.github/actions/setup-enterprise
       with:
         github-app-name: 'grafana-ci-bot'
-    - run: yarn install --immutable --check-cache
+    - name: Setup Node.js
+      uses: ./.github/actions/setup-node
+    - name: Yarn install
+      run: yarn install --immutable
+      env:
+        # If the PR isn't from a fork then don't use the slower yarn checks
+        YARN_ENABLE_HARDENED_MODE: ${{ github.event.pull_request.head.repo.fork == false && '1' || '0' }}
+        PUPPETEER_SKIP_DOWNLOAD: true
+        CYPRESS_INSTALL_BINARY: 0
     - run: yarn run test:ci
       env:
-        TEST_MAX_WORKERS: 2
-        TEST_SHARD: ${{ matrix.chunk }}
-        TEST_SHARD_TOTAL: 8
+        TEST_MAX_WORKERS: 4
+        TEST_SHARD: ${{ matrix.shard }}
+        TEST_SHARD_TOTAL: ${{ matrix.total }}
 
   frontend-decoupled-plugin-tests:
-    needs: detect-changes
+    needs:
+      - detect-changes
     if: needs.detect-changes.outputs.changed == 'true'
-    runs-on: ubuntu-latest-8-cores
+    runs-on: ubuntu-x64-large
     name: "Decoupled plugin tests"
     steps:
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
-    - uses: actions/setup-node@v4
-      with:
-        node-version-file: '.nvmrc'
-        cache: 'yarn'
-        cache-dependency-path: 'yarn.lock'
-    - run: yarn install --immutable --check-cache
+    - name: Setup Node.js
+      uses: ./.github/actions/setup-node
+    - name: Yarn install
+      run: yarn install --immutable
+      env:
+        # If the PR isn't from a fork then don't use the slower yarn checks
+        YARN_ENABLE_HARDENED_MODE: ${{ github.event.pull_request.head.repo.fork == false && '1' || '0' }}
+        PUPPETEER_SKIP_DOWNLOAD: true
+        CYPRESS_INSTALL_BINARY: 0
     - run: yarn run plugin:test:ci
 
   frontend-packages-unit-tests:
-    needs: detect-changes
+    needs:
+      - detect-changes
     if: needs.detect-changes.outputs.changed == 'true'
-    runs-on: ubuntu-latest-8-cores
+    runs-on: ubuntu-x64-large
     name: "Packages unit tests"
     steps:
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
-    - uses: actions/setup-node@v4
-      with:
-        node-version-file: '.nvmrc'
-        cache: 'yarn'
-        cache-dependency-path: 'yarn.lock'
-    - run: yarn install --immutable --check-cache
+    - name: Setup Node.js
+      uses: ./.github/actions/setup-node
+    - name: Yarn install
+      run: yarn install --immutable
+      env:
+        # If the PR isn't from a fork then don't use the slower yarn checks
```

nit: remove double negative
```suggestion
        # If the PR is from a fork then use the slower yarn checks
```